### DISCUSSION TITLE
Do ER diagrams commit anything to the DB?

### DISCUSSION THREAD

**Discussion by Michael-Dawson-Sage** [dbeaver/dbeaver#18033] [Q&A]
Hi. This is a very newb question. I'm using the "Create new ER diagram" functionality to visualize the relationships in an existing database. If I draw any relationships onto this diagram does it write anything back to the DB?
Or are the ER views just for visualisation only?

The reason I ask is I used to use this type of functionality in Access and visual relationships would be used to enforce referential integrity. I don't want to affect my DB, I just want to draw relationship pictures.

Thanks.

**Comment by mkgrgis** [comment]
> Do ER diagrams commit anything to the DB?

No. I have never seen any changes in DB corresponding to moving or changing elements of ERD. From ERD by ctrl You can find forms of "real" objects of database.

**Comment by Destrolaric** [comment]
@Michael-Dawson-Sage Hi, no ER diagram functionality doesn't make any changes in the database.

**Comment by Stilmant** [comment]
Thanks for the question because all this still not clear to me. 
in DBeaver Enterprise and Ultimate Editions it seems to be an 'Edit' mode in the ERD that allows to make changes saved/applied onto the database. (https://www.youtube.com/watch?v=dCm0Pva1Yaw)
With basic Edition you can have a ERD to visualize the database or to create some custom ERD diagrams for documentation purpose.  
What is strange to me is that relations I created in custom ERD diagrams are displayed when I generate a fresh ERD diagram of an database. (ER Diagram when you select a database and you go to ER Diagram tab beside Properties tab) 
THis confuse me a lot...

**Reply by Stilmant** [reply]
I understand what it is confusing me here is 'virtual foreign keys' that are maybe foreign key saved only locally and not in the database.
But why if I create one in a specific ERD (done for a test) that virtual relation appears in all ERD then? (it is acting like a virtual commit)

Also I believe there is no much documentation about 'virtual foreign keys' in dbeaver ? and is there are used (having impacts) in other place that ERD diagrams? 
Thanks.

**Comment by Stilmant** [comment]
Also this
![image](https://user-images.githubusercontent.com/149209/204258260-dd3e36a7-9cd6-4520-908d-7541f2dcd797.png)
displayed in the standard edition after editing an ERD (_as simple as rearranging a bit the diagram_)  the "persist these changes in the database" also confuse me..

**Reply by ikoif** [reply]
I was also confused once I saw this prompt. I thought changes were changed only locally and the docs (at the time of writing this comment) have no mention or details about the "Save" functionality - https://dbeaver.com/docs/dbeaver/ER-Diagrams/#diagram-export
